clients/spark/core/src/main/scala/io/treeverse/gc/UncommittedGarbageCollector.scala

```diff
@@ -157,13 +157,28 @@ object UncommittedGarbageCollector {
 
         var uncommittedDF =
           if (uncommittedGCRunInfo.uncommittedLocation != "") {
-            val uncommittedLocation =
-              ApiClient
-                .translateURI(new URI(uncommittedGCRunInfo.uncommittedLocation), storageType)
-                .toString
-            spark.read.parquet(uncommittedLocation)
+            try {
+              val uncommittedLocation =
+                ApiClient
+                  .translateURI(new URI(uncommittedGCRunInfo.uncommittedLocation), storageType)
+                  .toString
+              spark.read.parquet(uncommittedLocation)
+            } catch {
+              // Backwards compatibility with lakefs servers that return address even when there's no uncommitted data
+              case e: org.apache.spark.sql.AnalysisException =>
+                if (!e.message.contains("Path does not exist")) {
+                  throw e
+                }
+                println(
+                  "WARN: Uncommitted file list (" + uncommittedGCRunInfo.uncommittedLocation + ") not found - assuming no uncommitted data"
+                )
+                // in case of no uncommitted entries
+                spark.emptyDataFrame.withColumn("physical_address", lit(""))
```

Comment:
we can just fall throw or change the check so in case there is no path or no data we return an empty data frame